tests/epics/demo/test_demo.py

```diff
@@ -199,6 +199,28 @@ async def test_retrieve_mock_and_assert(mock_mover: demo.Mover):
     )
 
 
+async def test_mock_with_Mock():
```

Comment:
Nit: This name is a bit too implementation focussed IMO, something like `test_mocks_in_device_share_parent` would be more descriptive of the usecase